packages/studio-web/src/app/upload/upload.component.ts

```diff
@@ -38,7 +37,17 @@ import { HttpErrorResponse } from "@angular/common/http";
 })
 export class UploadComponent implements OnInit {
   langs$ = this.rasService.getLangs$().pipe(
-    map((langs: Array<SupportedLanguage> | HttpErrorResponse) => {
+    catchError((err, caught) => {
```

Comment:
This is already done in ras.service which is where I think it should be implemented.

Reply:
It would be better if we could explicitly subscribe to this observable and *catch* the error

Reply:
Now that I think of it, the real problem here is that this error is *not* recoverable at all!  You have to reload the page to make it work again.  I think this is a special case that can't be treated like the others, we shouldn't even be showing the application given that it absolutely will not work in this case.

Reply:
I changed this.  `langs` really has no business being an Observable anyway, it isn't a stream of data.  Also, we need to handle more HTTP errors than just failures to connect, and these *are* specific to different situations.  To make the code DRY I consolidated the reporting into a single method.  I think it is all much cleaner now, and we may even be able to recover from `langs` if the API comes back on line (which we couldn't do when it was an Observable referenced directly in the template code)

Reply:
indeed, we can now actually recover from `langs` failing, although just as `und` for the moment